deployer/src/deployment/deploy_layer.rs

```diff
@@ -558,23 +563,39 @@ mod tests {
     impl SecretGetter for StubSecretGetter {
         type Err = std::io::Error;
 
-        async fn get_secrets(&self, _service_id: &Uuid) -> Result<Vec<Secret>, Self::Err> {
+        async fn get_secrets(&self, _service_id: &Ulid) -> Result<Vec<Secret>, Self::Err> {
             Ok(Default::default())
         }
     }
 
     #[derive(Clone)]
     struct StubResourceManager;
 
-    #[async_trait::async_trait]
+    #[async_trait]
     impl ResourceManager for StubResourceManager {
         type Err = std::io::Error;
 
-        async fn insert_resource(&self, _resource: &Resource) -> Result<(), Self::Err> {
-            Ok(())
+        async fn insert_resources(
+            &mut self,
+            _resource: Vec<shuttle_proto::resource_recorder::record_request::Resource>,
+            _service_id: &ulid::Ulid,
+            _claim: Claim,
+        ) -> Result<ResultResponse, Self::Err> {
+            Ok(ResultResponse {
+                success: true,
+                message: "dummy impl".to_string(),
+            })
         }
-        async fn get_resources(&self, _service_id: &Uuid) -> Result<Vec<Resource>, Self::Err> {
-            Ok(Vec::new())
+        async fn get_resources(
+            &mut self,
+            _service_id: &ulid::Ulid,
+            _claim: Claim,
+        ) -> Result<ResourcesResponse, Self::Err> {
+            Ok(ResourcesResponse {
+                success: true,
+                message: "dummy impl".to_string(),
+                resources: Vec::new(),
+            })
```

Comment:
They are not used but we need them because they're implement for the `StubResourceManager`, which is used as a mock to instantiate a `DeploymentManager`, which is used in multiple places in the tests.

Reply:
Ahh, the return type changed :man_facepalming: